### DISCUSSION TITLE
Remove file doc block

### DISCUSSION THREAD

**Discussion by escopecz** [rectorphp/rector#7058] [Q&A] [answered]
I'm wondering how to remove file doc blocks like this one:
```php
<?php

/*
 * @copyright   2014 Mautic Contributors. All rights reserved
 * @author      Mautic
 *
 * @link        http://mautic.org
 *
 * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
 */

use ...
```
At first I tried to use [RemoveAnnotationRector](https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#removeannotationrector) with this config:
```php
$services->set(RemoveAnnotationRector::class)->configure(['copyright', 'author', 'license', 'link']);
```

But it does not catch it as it's [looking only for classes, methods, properties and constants](https://github.com/rectorphp/rector/blob/main/rules/DeadCode/Rector/ClassLike/RemoveAnnotationRector.php#L58). Also the file doc blocks do not start with `/**` but with `/*`. Is there any way how can rector find those and remove them? Or any direction how to write such rule would be appreciated.

**Comment by TomasVotruba** [comment, marked as answer]
Hi Honza,

in AST the docs do not exist on their own. They are automatically bound to next node, that can be anything here.
The same way attributes do not exist without class/property etc.

To taret the file docs, it's better to go with custom PHP CS Fixer rule and subscribe to `T_DOC`/`COMMENT` tokens.
Detect the content by obvious markers and remove it.

**Reply by escopecz** [reply]
Thanks for the direction! This worked: https://stackoverflow.com/questions/40244960/php-cs-fixer-how-to-remove-header-docblock-comments

**Reply by TomasVotruba** [reply]
Yes, I used that one before :+1:
